json-schema-provider/src/test/java/io/confluent/kafka/schemaregistry/json/JsonSchemaTest.java

```diff
@@ -324,6 +328,36 @@ public void testRecursiveSchema() {
     assertEquals(0, diff.size());
   }
 
+  @Test
+  public void testParseSchema() {
+    SchemaProvider jsonSchemaProvider = new JsonSchemaProvider();
+    ParsedSchema parsedSchema = jsonSchemaProvider.parseSchemaOrElseThrow(recordSchemaString,
+            new ArrayList<>(), false);
+    Optional<ParsedSchema> parsedSchemaOptional = jsonSchemaProvider.parseSchema(recordSchemaString,
+            new ArrayList<>(), false);
+
+    assertNotNull(parsedSchema);
+    assertTrue(parsedSchemaOptional.isPresent());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testParseSchemaThrowException() {
+    String invalidSchemaString = "{\"properties\": {\n"
+            + "  \"string\": {\"type\": \"str\"}\n"
+            + "  }"
+            + "  \"additionalProperties\": false\n"
+            + "}";
+
+    SchemaProvider jsonSchemaProvider = new JsonSchemaProvider();
+    Optional<ParsedSchema> parsedSchema = jsonSchemaProvider.parseSchema(invalidSchemaString,
+            new ArrayList<>(), false);
+    assertFalse(parsedSchema.isPresent());
```

Comment:
Test would pass if this call here throws `IllegalArgumentException`. Can we separate this into two separate tests ?

Reply:
Thanks for the review @dragosvictor . Addressed.